tests/scenarios/v2-addon-dev-test.ts

```diff
@@ -0,0 +1,214 @@
+import path from 'path';
+import { appScenarios, baseV2Addon } from './scenarios';
+import { PreparedApp } from 'scenario-tester';
+import QUnit from 'qunit';
+import merge from 'lodash/merge';
+import { pathExistsSync, readJsonSync } from 'fs-extra';
+
+const { module: Qmodule, test } = QUnit;
+
+/**
+ * The type of addon this is testing with only works in
+ * ember-source@3.25+
+ */
+appScenarios
+  .skip('lts_3_16')
+  .skip('lts_3_24')
+  .map('v2-addon-dev', async project => {
+    let addon = baseV2Addon();
+    addon.pkg.name = 'v2-addon';
+    addon.pkg.files = ['dist'];
+    addon.pkg.exports = {
+      './*': './dist/*',
+      './addon-main.js': './addon-main.js',
+      './package.json': './package.json',
+    };
+    addon.pkg.scripts = {
+      build: './node_modules/rollup/dist/bin/rollup -c ./rollup.config.mjs',
+    };
+
+    merge(addon.files, {
+      'babel.config.json': `
+        {
+          "presets": [
+            ["@babel/preset-env", {
+              "targets": ["last 1 firefox versions"]
+            }]
+          ],
+          "plugins": [
+            "@embroider/addon-dev/template-colocation-plugin",
+            ["@babel/plugin-proposal-decorators", { "legacy": true }],
+            [ "@babel/plugin-proposal-class-properties" ]
+          ]
+        }
+      `,
+      'rollup.config.mjs': `
+        import { babel } from '@rollup/plugin-babel';
+        import { Addon } from '@embroider/addon-dev/rollup';
+
+        const addon = new Addon({
+          srcDir: 'src',
+          destDir: 'dist',
+        });
+
+        export default {
+          output: addon.output(),
+
+          plugins: [
+            addon.publicEntrypoints([
+              '**/*.js',
+              'components/demo/out.hbs',
```

Comment:
Similar to the meaning in imports, we need a way to talk about template files, so people need to be allowed to put explicit `.hbs` here like this. But that can't mean the same thing as the template-only component. The template-only component is a JS file -- even if that JS file is synthesized by the build, rather than typed by the author.